src/backend/SwapChain.cpp

```diff
@@ -34,17 +34,23 @@ namespace backend {
         return device;
     }
 
-    void SwapChainBase::Configure(nxt::TextureFormat format, uint32_t width, uint32_t height) {
+    void SwapChainBase::Configure(nxt::TextureFormat format, nxt::TextureUsageBit allowedUsage, nxt::TextureUsageBit initialUsage, uint32_t width, uint32_t height) {
         if (width == 0 || height == 0) {
             device->HandleError("Swap chain cannot be configured to zero size");
             return;
         }
+        if ((initialUsage & allowedUsage) == nxt::TextureUsageBit::None) {
```

Comment:
I think it should check that initialUsage has exactly one bit, making it:

Then this check becomes `if (!(HasZeroOrOneBits(initialUsage) && (initialUsage & allowedUsage)))` or something like that.